### PR TITLE
Contrib get repository tokens

### PR DESCRIPTION
**Pull request by HaraldMuellerNey**
## Description

Adding a small python script to get tokens via SCC API to setup custom repository.
The script requires user/password of the ORGA mirror credentials on command line.
You can provide various "strings" (URL Snippets).
For  each matching repository the scripts print one line providing:
_full url path,TOKEN_

Fixes # none

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [x] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [ N/A] I have verified that my code follows RMT's coding standards with `rubocop`.
- [x] I have reviewed my own code and believe that it's ready for an external review.
- [x] I have provided comments for any hard-to-understand code.
- [ N/A] I have documented the `MANUAL.md` file with any changes to the user experience.
- [x] RMT's test coverage remains at 100%.
- [N/A] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

### CODE REVIEW

File: contrib/get_repository_tokens.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:1]: **Python2 EOL**

```suggestion
#!/usr/bin/env python3
```

File: contrib/get_repository_tokens.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                        
+
+def get_token(match_repo_url):
+    resp = s.get("https://scc.suse.com/connect/organizations/repositories", headers={'Accept':'application/vnd.scc.suse.com.v4+json','cache-control':'no-cache'})
+    repositories = []
+
+    while True:
+        for repository in resp.json():
+            if match_repo_url in repository['url'].split("?")[0]:
+                print repository['url'].split("?")[0]+","+repository['url'].split("?")[-1]
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:29]: **Python2 EOL**
```suggestion
                print(repository['url'].split("?")[0]+","+repository['url'].split("?")[-1])
```

File: contrib/get_repository_tokens.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        rels = get_rels(resp.headers)
+        if not 'next' in rels:
+            break
+        resp = s.get(rels['next'])
+
+def print_USAGE():
+    print 'Usage:'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:36]: **Python2 EOL**

```suggestion
    print('Usage:')
```

File: contrib/get_repository_tokens.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    print '\t'+self+' -u <username> -o <password> <part of repository URL> [... <part of repository URL>]'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:37]: **Python2 EOL**

Usage instruction is wrong. The password flag needs to be `-p`.

```suggestion
    print('\t'+self+' -u <username> -p <password> <part of repository URL> [... <part of repository URL>]')
```

File: contrib/get_repository_tokens.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    print '\tUse the SCC mirror credentials of your organization for user and password'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:38]: **Python2 EOL**

```suggestion
    print('\tUse the SCC mirror credentials of your organization for user and password')
```

File: contrib/get_repository_tokens.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    print '\tProvide strings to match against repository URL'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:39]: **Python2 EOL**
```suggestion
    print('\tProvide strings to match against repository URL')
```

File: contrib/get_repository_tokens.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    print '\tThe script provides all matches, one per line, full URL and TOKEN separated by ",".'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:40]: **Python2 EOL**

```suggestion
    print('\tThe script provides all matches, one per line, full URL and TOKEN separated by ",".')
```

File: contrib/get_repository_tokens.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    for opt, arg in opts:
+        if opt in ('-h','--help'):
+            print_USAGE()
+            sys.exit()
+        elif opt in ("-u", "--user"):
+            user = arg
+        elif opt in ("-p", "--password"):
+            password = arg
+    if not user or not password:
+        print 'ERROR: missing user or password of your mirror credentials'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:59]: **Python2 EOL**
```suggestion
        print('ERROR: missing user or password of your mirror credentials')
```

File: contrib/get_repository_tokens.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        print_USAGE()
+        sys.exit()
+    if not args:
+        print 'ERROR: no strings to match against repository URL'
```
Comment (ngetahun) [inline contrib/get_repository_tokens.py:63]: **Python2 EOL**
```suggestion
        print('ERROR: no strings to match against repository URL')
```

**Review by ngetahun — CHANGES_REQUESTED** [review]

**Review by ngetahun — APPROVED** [review]

**Comment by Sergeykot** [comment]
@HaraldMuellerNey @ngetahun should I merge and release it?

**Comment by HaraldMuellerNey** [comment]
It is ready to merge from my end.

**Comment by ngetahun** [comment]
@Sergeykot Can you hold off on the merge? We still need to release 2.6.9.

**Comment by Sergeykot** [comment]
@ngetahun since we do not package the script we can merge it independently n the release.
@HaraldMuellerNey do you prefer this script to be in Python or we can also merge our Ruby port?

**Comment by HaraldMuellerNey** [comment]
Feel free to convert it - I wrote it for myself and just thought it might be helpful for others.

**Pull request merged**